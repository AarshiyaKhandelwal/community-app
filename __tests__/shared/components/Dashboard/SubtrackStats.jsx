import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import ReactDOM from 'react-dom';
import SubtrackStats from 'components/Dashboard/SubtrackStats';

/* global document, window */

const mockData = {
  subtracks: [
    {
      track: 'DEVELOP',
      subtrack: 'CODE',
      stat: 1000,
    },
    {
      track: 'DESIGN',
      subtrack: 'DESIGN',
      stat: 1000,
    },
  ],
  handle: 'handle',
};

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  renderer.render((
    <SubtrackStats {...mockData} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});


describe('handler resize', () => {
  const container = document.createElement('div');
  let instance;

  beforeEach(() => {
    instance = ReactDOM.render(<SubtrackStats {...mockData} />, container);
  });

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(container);
  });

  test('resize to 1920', () => {
    window.resizeTo(1920, 1280);
    expect(instance.state.width).toEqual(1920);
  });

  test('resize to 1280', () => {
    window.resizeTo(1280, 960);
    expect(instance.state.width).toEqual(1280);
  });
});
