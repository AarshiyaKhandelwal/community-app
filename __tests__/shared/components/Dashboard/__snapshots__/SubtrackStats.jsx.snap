// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-Dashboard-SubtrackStats-___styles__subtrack-stats___2XEE_"
>
  <div
    className="src-shared-components-Dashboard-SubtrackStats-___styles__ratings___2pnzh"
  >
    <div
      className="src-shared-components-Dashboard-SubtrackStats-___styles__tracks___2UnlG"
    >
      <div
        className="src-shared-components-Dashboard-SubtrackStats-___styles__slide-wrapper___1xP_x"
      >
        <a
          className="src-shared-components-Dashboard-SubtrackStats-___styles__track___1MS06"
          href="https://www.topcoder-dev.com/members/handle/details/?track=DEVELOP&subTrack=undefined"
        >
          <div
            className="src-shared-components-Dashboard-SubtrackStats-___styles__flex-wrapper___1JpvN"
          >
            <p
              className="src-shared-components-Dashboard-SubtrackStats-___styles__subtrack___2TXRB"
              title=""
            />
            <p
              className="src-shared-components-Dashboard-SubtrackStats-___styles__rating___9uUVx"
              style={
                Object {
                  "color": "#69C329",
                }
              }
            >
              1000
              <span
                style={
                  Object {
                    "backgroundColor": "#69C329",
                  }
                }
              />
            </p>
            <p />
          </div>
        </a>
        <a
          className="src-shared-components-Dashboard-SubtrackStats-___styles__track___1MS06"
          href="https://www.topcoder-dev.com/members/handle/details/?track=DESIGN&subTrack=undefined"
        >
          <div
            className="src-shared-components-Dashboard-SubtrackStats-___styles__flex-wrapper___1JpvN"
          >
            <p
              className="src-shared-components-Dashboard-SubtrackStats-___styles__subtrack___2TXRB"
              title=""
            />
            <p
              className="src-shared-components-Dashboard-SubtrackStats-___styles__rating___9uUVx"
              style={
                Object {
                  "color": "#21B2F1",
                }
              }
            >
              1000
            </p>
            <p />
          </div>
        </a>
      </div>
    </div>
    <div
      className="src-shared-components-Dashboard-SubtrackStats-___styles__responsive-carousel___3atTT"
    >
      <Slider
        centerMode={true}
        dots={false}
        infinite={false}
        nextArrow={
          <NavButton
            currentSlide={null}
            slideCount={null}
          />
        }
        prevArrow={
          <NavButton
            currentSlide={null}
            slideCount={null}
          />
        }
        slidesToScroll={1}
        slidesToShow={1}
        speed={500}
      >
        <div>
          <div
            className="src-shared-components-Dashboard-SubtrackStats-___styles__slide-wrapper___1xP_x"
          >
            <a
              className="src-shared-components-Dashboard-SubtrackStats-___styles__track___1MS06"
              href="https://www.topcoder-dev.com/members/handle/details/?track=DEVELOP&subTrack=undefined"
            >
              <div
                className="src-shared-components-Dashboard-SubtrackStats-___styles__flex-wrapper___1JpvN"
              >
                <p
                  className="src-shared-components-Dashboard-SubtrackStats-___styles__subtrack___2TXRB"
                  title=""
                />
                <p
                  className="src-shared-components-Dashboard-SubtrackStats-___styles__rating___9uUVx"
                  style={
                    Object {
                      "color": "#69C329",
                    }
                  }
                >
                  1000
                  <span
                    style={
                      Object {
                        "backgroundColor": "#69C329",
                      }
                    }
                  />
                </p>
                <p />
              </div>
            </a>
            <a
              className="src-shared-components-Dashboard-SubtrackStats-___styles__track___1MS06"
              href="https://www.topcoder-dev.com/members/handle/details/?track=DESIGN&subTrack=undefined"
            >
              <div
                className="src-shared-components-Dashboard-SubtrackStats-___styles__flex-wrapper___1JpvN"
              >
                <p
                  className="src-shared-components-Dashboard-SubtrackStats-___styles__subtrack___2TXRB"
                  title=""
                />
                <p
                  className="src-shared-components-Dashboard-SubtrackStats-___styles__rating___9uUVx"
                  style={
                    Object {
                      "color": "#21B2F1",
                    }
                  }
                >
                  1000
                </p>
                <p />
              </div>
            </a>
          </div>
        </div>
      </Slider>
    </div>
  </div>
</div>
`;
